cardano-api/src/Cardano/Api/ScriptData.hs

```diff
@@ -131,21 +136,30 @@ instance ToCBOR ScriptData where
 instance FromCBOR ScriptData where
   fromCBOR = fromPlutusData <$> decode @Plutus.Data
 
-hashScriptData :: ScriptData -> Hash ScriptData
+type HashableScriptData = WithCBOR ScriptData
+
+pattern AsHashableScriptData :: AsType (WithCBOR ScriptData)
+pattern AsHashableScriptData = AsWithCBOR AsScriptData
```

Comment:
I thought a pattern might make it feel more like all the other `AsType`s, but I'm starting to rethink it